plugins/ssr/ssrequality.ml

```diff
@@ -399,15 +399,16 @@ let pirrel_rewrite ?(under=false) ?(map_redex=id_map_redex) pred rdx rdx_ty new_
       let c1' = Global.constant_of_delta_kn (Constant.canonical (Constant.make2 mp l')) in
       sigma, EConstr.of_constr (mkConst c1')
   in
-  let proof = EConstr.mkApp (elim, [| rdx_ty; new_rdx; pred; p; rdx; c |]) in
+  let proof = EConstr.mkApp (elim, [| rdx_ty; new_rdx; pred |]) in
   (* We check the proof is well typed *)
   let sigma, proof_ty =
     try Typing.type_of env sigma proof with
     | Pretype_errors.PretypeError (env, sigma, te) -> raise (PRtype_error (Some (env, sigma, te)))
     | e when CErrors.noncritical e -> raise (PRtype_error None)
   in
+  let proof = EConstr.mkApp (proof, [| p; rdx; c |]) in
   debug_ssr (fun () -> Pp.(str"pirrel_rewrite: proof term: " ++ pr_econstr_env env sigma proof));
-  debug_ssr (fun () -> Pp.(str"pirrel_rewrite of type: " ++ pr_econstr_env env sigma proof_ty));
+(*   debug_ssr (fun () -> Pp.(str"pirrel_rewrite of type: " ++ pr_econstr_env env sigma proof_ty)); *)
   Proofview.tclORELSE (refine_with
     ~first_goes_last:(not !ssroldreworder || under) ~with_evars:under (sigma, proof))
```

Comment:
Before the pull request, `sigma` was the evar map that resulted from typing `proof`. This is no longer the case after the pull request. Does it matter?

Reply:
Hopefully it should not because there are no universes involved. But maybe I am wrong.

Reply:
I don't think it matters, but the debug print can stay (modulo fixing the string: you can also print proof, and write a comment saying why you type the partial app....)